
\documentclass[journal,transmag]{IEEEtran}

\usepackage{cite}
\usepackage{color}
\usepackage[table]{xcolor}
\usepackage{amsmath}
\usepackage{multirow}
\usepackage{graphicx}
\usepackage{threeparttable}
\ifCLASSOPTIONcompsoc
\usepackage[caption=false,font=normalsize,labelfon
t=sf,textfont=sf]{subfig} \else
\usepackage[caption=false,font=footnotesize]{subfig}
\fi
 
\newcommand\MYhyperrefoptions{bookmarks=true,bookmarksnumbered=true,
pdfpagemode={UseOutlines},plainpages=false,pdfpagelabels=true,
colorlinks=true,linkcolor={black},citecolor={black},urlcolor={black},
pdftitle={Bare Demo of IEEEtran.cls for Computer Society Journals},%<!CHANGE!
pdfsubject={Typesetting},%<!CHANGE!
pdfauthor={Michael D. Shell},%<!CHANGE!
pdfkeywords={Computer Society, IEEEtran, journal, LaTeX, paper,
             template}}
\hyphenation{op-tical net-works semi-conduc-tor}


\begin{document}
\title{Analysis of Co-Associated Transcription Factors \\ via Ordered Adjacency
Differences \\ on Motif Distribution}

\author{Gaofeng Pan,
        Jijun Tang, and Fei Guo, % <-this % stops a space
\IEEEcompsocitemizethanks{ \IEEEcompsocthanksitem G. Pan, F. Guo are
with School of Computer Science and Technology, Tianjin University, No.135, Yaguan Road, Tianjin Haihe Education Park, Tianjin, P.R.China.\protect\\
E-mail: fguo@tju.edu.cn, F. Guo is the corresponding author.
\IEEEcompsocthanksitem J. Tang is with School of Computational Science and Engineering, University of South Carolina, Columbia, USA.}% <-this % stops a space
\thanks{Manuscript received April 19, 2005; revised September 17, 2014.}}

\markboth{Journal of \LaTeX\ Class Files,~Vol.~13, No.~9, September~2014}%
{Shell \MakeLowercase{\textit{et al.}}: Bare Advanced Demo of
IEEEtran.cls for Journals}

\IEEEtitleabstractindextext{
\begin{abstract}


Transcription factors (TFs) bind to specific DNA sequences or
motifs, and interactions between TFs and DNA are elementary to the
regulation of transcription. Generally, the gene is regulated by a
combination of TFs in close proximity. Analysis of co-TFs is an
important problem in understanding the mechanism of transcriptional
regulation. Recently, ChIP-seq in mapping TF binding sites provide a
large amount of experimental data to analyze co-TFs. Several studies
shows that if two TFs are co-associated, the relative distance of
each TF with respect to the other exhibits a peak-like distribution.
In order to analyze co-TFs, we develop a novel method to evaluate
the associated situation between TFs. We design an adjacency score
for motif analysis, which can illustrate ordered adjacency
differences and co-TF binding affinity for candidate motifs. For all
candidate motifs, we calculate corresponding adjacency scores, and
then list descending-order motifs corresponding to their motif
scores. From these lists, we can find co-TF binding affinity for
candidate motifs. On $13$ ChIP-seq datasets, our method obtains best
AUC results on four datasets, $0.8828$ for E2F1, $0.8466$ for ESRRB,
$0.8818$ for KLF4 and $0.9432$ for NMYC. CENTDIST produces best AUC
results on eight datasets, and CORE\_TF promBG under promoter
background uses enriched regions with size $200$ gets the best AUC
result on one datasets. AUC results of our method on $13$ datasets
are all above $0.8$, and the most accurate result is $0.9611$
archived on CMYC.

\end{abstract}


\begin{IEEEkeywords}

transcription factors, co-TFs, ChIP-seq, peak-like distribution,
ordered adjacency difference, AUC.

\end{IEEEkeywords}}


\maketitle \IEEEdisplaynontitleabstractindextext
\IEEEpeerreviewmaketitle



\section{Introduction}

Transcription factors (TFs) recognize specific DNA sequences near
promoter regions of genes. TFs binding specificities play key roles
in gene regulatory network architectures and functions \cite{mybibfile:tfs}. TFs bind to
specific DNA sequences or motifs in our genome, and interactions
between TFs and DNA are elementary to the regulation of
transcription. Therefore, the ability of detecting TF binding sites
throughout genomes is necessary to reflect gene regulation and infer
regulatory networks \cite{mybibfile:proteindna} \cite{mybibfile:chipseqtfs}.


Generally, the gene is not regulated by only one single TF, but
instead by a combination of TFs in close proximity \cite{mybibfile:multiregulator}. TFs co-localize
and collaborate together, known as co-associated TFs (co-TFs) of
each other. Analysis of co-TFs is an important problem in
understanding the mechanism of transcriptional regulation \cite{mybibfile:transregulation} \cite{mybibfile:tfscoregu} \cite{mybibfile:tfcoexpress}. Recently,
ChIP-seq in mapping TF binding sites provide a large amount of
experimental data to analyze co-TFs \cite{mybibfile:chipseqqa} \cite{mybibfile:chipchipseq} \cite{mybibfile:chipvivo}. There exist many technologies
in mapping TF binding sites to identify novel co-TFs.


Motif Enrichment Analysis (MEA) uses enrichment information of known
motifs in the regions of genes to determine whether DNA-binding
transcription factors have function on a set of genes
\cite{mybibfile:mea}. There are some MEA methods for motif
enrichment analysis with difference feature and performance, such as
ConTra \cite{mybibfile:contra}, PASTAA \cite{mybibfile:pastaa},
SpaMo \cite{mybibfile:spamo}, CEAS \cite{mybibfile:ceas}, CORE\_TF
\cite{mybibfile:coretf} and CENTDIST \cite{mybibfile:centdist}.


ConTra (conserved TFBSs) can do motif enrichment analysis in the
promoters of genes. With gene sequences of several species, ConTra
checks whether motif binding sites are conserved in genes. For a
list of motifs, scores reflect their binding situations to the
promoter. PASTAA (Predict ASsociated Transcription factors from
Annotated Affinities) uses binding affinities of TF to detect
binding motifs. In PASTAA, all genes are ranked according to their
predicted affinity for a given TF and their association with a given
category separately. ConTra and PASTAA can only do enrichment
analysis on promoter region but not genomic region.


SpaMo (spaced motif analysis) is able to infer interactions between
the specific TF and TFs bound at near sites on the DNA sequence.
This method can get motif spacing information facilitating the
understanding of individual TF complex structures. Unlike other
motif enrichment analysis method, SpaMo analyzes the enrichment of
motif spacings instead of occurrences. CEAS (cis-regulatory element
annotation system) can do motif finding and enrichment analysis.
Given ChIPed regions and a motif, CEAS counts the number of hits,
where the score of motif is greater than a cutoff, both in the ChIP
region and in the whole genome, then report and rank motifs
according to the binomial test P-value. CORE\_TF (Conserved and
Over-REpresented Transcription Factor binding sites) use both
sequence conservation-based approach and PWM approach. The
combination of these two approaches can reduce false predictions
when identify TF binding sites. With a input dataset, CORE\_TF
subsequently scans individual promoters for cross-species
conservation, then employs PWM matrices. CENTDIST use the property
center distribution, that two TFs are co-associated when the
relative distance between them exhibits a peak-like distribution
\cite{mybibifle:centdistgenomic,mybibfile:centdistsignal,mybibfile:centdistnucleo}.
For a input ChIP-seq dataset, CENTDIST scans for the occurrence of
bindings around the peak point, then scores the imbalanced
distribution of motifs. SpaMo, CEAS, CORE\_TF and CENTDIST can
accept genomic regions as input dataset and do analysis of them.


Several studies shows that if two TFs are co-associated, their
ChIP-seq peaks are not only in close proximity with each other, but
the relative distance of each TF with respect to the other exhibits
a peak-like distribution. In order to analyze co-TFs, we develop a
novel method to evaluate the associated situation between TFs.
First, we design the sequence-specific binding score for
representing patterns in biological sequences. Then, we produce
ordered adjacency scores based on a novel descending-order matrix.
These two scores reflect the difference information between two
adjacent regions to analyze the tendency of binding affinity between
motif and DNA sequences. For all candidate motifs, we calculate
corresponding adjacency scores for each dataset, and then list
descending-order motifs corresponding to their motif scores. From
these lists, we can find co-TF binding affinity for candidate
motifs. On $13$ ChIP-seq datasets, our method obtains best AUC
results on four datasets, $0.8828$ for E2F1, $0.8466$ for ESRRB,
$0.8818$ for KLF4 and $0.9432$ for NMYC. CENTDIST produces best AUC
results on eight datasets, and CORE\_TF promBG under promoter
background uses enriched regions with size $200$ gets the best AUC
result on one datasets. AUC results of our method on $13$ datasets
are all above $0.8$, and the most accurate result is $0.9611$
archived on CMYC.


\section{Method}


We can obtain a large amount of ChIPed TF's location data by
ChIP-seq \cite{mybibfile:chipseq} experiment. Locations of co-TFs
for a particular TF always enrich around this TF's location. One
problem is to identify co-TFs of ChIPed TF with a list of ChIP-seq
peaks, which map TFs on gene sequences. Assuming binding motifs of
candidate co-TFs are known, the approach to this challenge is motif
enrichment analysis.


In order to predict co-associated TFs, we develop a novel method to
evaluate the associated situation between TFs. We design an
adjacency score for motif analysis, which can illustrate ordered
adjacency differences and co-TF binding affinity for candidate
motifs.


\subsection{Sequence-Specific Binding Score}

DNA motif is denoted as the conservation feature of binding sequence
for a TF. We can use the common representation, Position Weight
Matrix (PWM) \cite{mybibfile:pwm}, for modeling DNA motif
computationally. It has great advantages for representing patterns
in biological sequences \cite{mybibfile:pwmadv}.


PWM models a $l$-bases motif as a $4 \times l$ matrix $\Theta$. The
entry $\Theta_{q,p}$ is the frequency of nucleotide $q$ ($q\in\{A,
C, G, T\}$) at position $p$, and all entries in each column of
matrix sum to $1$. Given a $l$-bases sequence $S$, $S[i]$ denotes
the base at position $i$, $\Theta_{S[i],i}$ denotes the probability
of nucleotide $S[i]$ at position $i$ under PWM matrix $\Theta$, and
$Pr[s|\Theta]=\prod_{i=1}^{l}\Theta_{s[i],i}$ denotes the
probability of producing sequence $S$ from matrix $\Theta$.


The PWM score is the log likelihood ratio of the probability
$Pr[s|\Theta]$, compared to a uniform $0$-markov model
\cite{mybibfile:markovmodel01,mybibfile:markovmodel02}. Given a
sequence $S$ and a PWM matrix $\Theta$ with length $l$, the PWM
score can be defined as follows.

\begin{equation}
\label{pwm_equation} s_{pwm} =
\sum\limits_{i=1}^{l}\log\Big(\frac{\Theta_{S[i],i}}{0.25}\Big)
\end{equation}
where $0.25$ is the probability under uniform model.


For a $l$-bases motif, we can calculate the nucleotides sequence
with the maximum or minimum PWM score. The maximum PWM score
$s_{max}$ can be defined as follows.

\begin{equation}
\label{pwm_max} s_{max} = \sum\limits_{i=1}^{l}
\log\Big(\frac{\max\{\Theta_{N,i} | N = A, C, G, T \}}{0.25}\Big)
\end{equation}
where $\max\{\Theta_{N,i} | N = A, C, G, T \}$ is the maximum
probability chosen in each column of PWM matrix.

The minimum PWM score $s_{min}$ can be defined as follows.

\begin{equation}
\label{pwm_min} s_{min} = \sum\limits_{i=1}^{l}
\log\Big(\frac{\min\{\Theta_{N,i} | N = A, C, G, T \}}{0.25}\Big)
\end{equation}
where $\min\{\Theta_{N,i} | N = A, C, G, T \}$ is the minimum
probability chosen in each column of PWM matrix.

We can use the linear transformation to normalize each PWM score
$s_{pwm}$ within the range of $[0,1]$, defined as follows.

\begin{equation}
\label{norm_pwm_equ} v = \frac{s_{pwm} - s_{min}}{s_{max} - s_{min}}
\end{equation}
where sequence-specific binding score $v$ represents the binding
affinity between motif and sequences.


The sequence-specific binding information can be shown in Figure
\ref{pwm_figure}. Figure \ref{motifpwm}) is the PWM matrix of motif
V\$MYOD\_01 in TRANSFAC database. This motif data is defined by five
functional elements in three genes of mouse, and each value in this
matrix represent the frequency of corresponding nucleotide at the
specific position of aligned sequences. For example, $0.2$ at cell
($A$, $1$) means that in aligned sequences, there exist $20\%$
sequences having nucleotide $A$ at position $1$.


Sequence logo \cite{mybibfile:seqlogo} is a graphical representation
of the sequence conservation. Figure \ref{seqlogo} is the sequence
logo of V\$MYOD\_01, plotted using Biopython
\cite{mybibfile:weblogo}. It depicts the consensus sequence and the
diversity of sequences. The relative size of each letter indicates
its frequency in aligned sequences. Biopython
\cite{mybibfile:biopython} is an open-source system to parse gene
data. We use the package of Biopython to parse TRANSFAC matrix
entries and calculate the PWM score.


\begin{figure*}[!htpb]
\centering \subfloat[Position Weight Matrix of V\$MYOD\_01]{
\includegraphics[width=5in]{image/motifpwm}
\label{motifpwm}} \hfil \subfloat[Sequence Logo of V\$MYOD\_01]{
\includegraphics[width=3.5in]{image/seqlogo}
\label{seqlogo}} \caption{The sequence-specific binding information
of motif V\$MYOD\_01 in TRANSFAC database.} \label{pwm_figure}
\end{figure*}



\subsection{Descending-Order Matrix}


The peak point set in the ChIP-seq map can be represented as $P =
\{p_1, p_2, ..., p_n\}$. DNA sequences are extracted from the $\pm
m$ bp region of every peak $p_i$ in the ChIP-seq data, and the
sequence set is constructed as $S = \{s_1, s_2, ..., s_n\}$. We
partition every extracted sequence into $b$ bins with respect to the
distance to the peak point, where each bin is of size $\frac{2
\times m}{b}$ bp.

For each motif, we scan all sub-sequences in each bin with the PWM
matrix $\Theta$. We use the normalized PWM score $v$ to construct a
$n \times b$ matrix $M_{s}$ with the descending-order column, and
binding scores of sub-sequences are uniformly put into corresponding
columns according to absolute distances between their bp positions
and peak point positions. We normalize the value of each cell by
using the maximum value in its corresponding column.


We analyze an example of extracting sequences and creating matrix
$M_{s}$ on GSM288345 dataset, shown in Figure \ref{scmatrix}. Figure
\ref{scmatrix} (left) represents the sequence set extracted from
mouse genome using ChIP-seq of NANOG. The middle point is
corresponding to the peak point, and the sequence is $2000$ bp
length including both left and right $1000$ bp regions. The position
$3,053,033$ is a peak point on GSM288345, and the sequence
$[3,052,033, 3,054,033]$ is extracted as the first sequence. Figure
\ref{scmatrix} (right) represents the matrix of normalized binding
scores, with the descending-order column.


\begin{figure*}[!htpb]
\centering \subfloat{
\includegraphics[width=7in]{image/geneseq}
\label{geneseq}} \caption{An example of extracting sequences and
creating matrix $M_{s}$ on GSM288345 dataset} \label{scmatrix}
\end{figure*}


\subsection{First-Order Adjacency Difference}

We calculate the difference between two adjacent columns in matrix
$M_{s}$, to analyzing the tendency of binding affinity between motif
and sub-sequences. We extract a pair of adjacent columns in each
region, and calculate first-order adjacency difference $f_{1}$
between each pair of adjacent cells in the same row, defined as
follows.

\begin{equation}
\label{one_order_diff} f_{1}(i, j) = w(i) ( M_{s}[i,j] -
M_{s}[i,j+1] )
\end{equation}
where $i$ is the index of each row, and $j$ and $j+1$ are indices of
columns.


Considering co-TFs distribution
\cite{mybibifle:centdistgenomic,mybibfile:centdistsignal,mybibfile:centdistnucleo},
they always appear near around peak points. Therefore, we use a
gamma distribution function \cite{mybibfile:gammadist} to weight the
$f_{1}$ score, which has large values at near regions and small
values at remote regions.

We sum top-$k$ $f_{1}$ values in each region and weight results by
the gamma distribution $g(j|c,\gamma)$ according to the region $j$.
Then, the total score of first-order adjacency difference $S_{1}$
can be defined as follows.

\begin{equation}
\label{score_f_equ} S_{1} = \sum\limits_{j=1}^{b-1} g(j|c,\gamma)
\sum\limits_{i=1}^{k} f_{1}(i, j)
\end{equation}


\subsection{Second-Order Adjacency Difference}

When defining above scores, we only consider equal possibility model
as the background model. However, CG/AT bias around ChIP-seq peak
points has unbalanced distribution. We define Second-Order Adjacency
Difference $f_{2}$ to reduce the effect of unbalanced CG/AT bias
noise, as follows.

\begin{equation}
\label{two_order_diff} f_{2}(i, j) = f_{1}(i, j) - f_{1}(i, j+1)
\end{equation}


A large or positive difference means a dense-binding region, but a
small or negative means a sparse-binding region. Therefore, we sum
top-$k$ $f_{2}$ values in each region and use sigmoid function
\cite{mybibfile:sigmoid} to normalize the difference of each region.
Then, the total score of second-order adjacency difference $S_{2}$
can be defined as follows.

\begin{equation}
\label{score_v_equ} S_{2} =
\frac{1}{b-2}\sum\limits_{j=1}^{b-2}\frac{1}{1 +
\frac{1}{e^{\Big(\sum_{i=1}^{k}f_{2}(i, j)\Big)}}}
\end{equation}



\subsection{Adjacency Score for Motif Analysis}

Combining above two difference scores, we can calculate the final
adjacency score $S_{motif}$ of motif $\Theta$ for motif analysis
around ChIPed points, defined as follows.

\begin{equation}
\label{motif_score_equ} S_{motif} = \omega_{1} \cdot S_{1} +
\omega_{2} \cdot S_{2}
\end{equation}
where $\omega_{1}$ and $\omega_{2}$ are weights of $S_{1}$ and
$S_{2}$.

For all candidate motifs, we calculate corresponding adjacency
scores for each dataset, and then list descending-order motifs
corresponding to their motif scores. From these lists, we can find
co-TF binding affinity for candidate motifs.



\section{Result}


We apply our method on several datasets, and use AUC to evaluate the
performance of results. Then, we analyze the ordered adjacency
difference defined by our method. Finally, we compare results of our
method with other existing methods, and find that our method improve
on some datasets.


\subsection{Data Set}

We use ChIP-seq map of TFs, genome sequence and motif matrix to
analyze co-TFs.


ChIP-seq data \cite{mybibfile:mms10} are mapping of $13$
transcription factors in mouse embryonic stem (ES) cells, shown in
Table \ref{bind_tf_families}. We test on $13$ transcription factors,
such as Nanog, Oct4, STAT3, Smad1, Sox2, Zfx, c-Myc, n-Myc, Klf4,
Esrrb, Tcfcp2l1, E2f1 and p300. Among these factors, p300 is
transcription regulator and others are sequence specific
transcription factors. From these TFs data, we use the chromosome
number and peak location in mouse (Mus musculus) genome.

\begin{table*}[!htpb]
\centering
\begin{threeparttable}
\renewcommand{\arraystretch}{1.3}
\caption{Motif family of TFs and co-TFs for ChIP-seq data.}
\label{bind_tf_families}
\begin{tabular}{|l|l|}
\hline
\textbf{ChIP-Seq Data} & \textbf{Motif Family of TFs and co-TFs} \\ \hline
Nanog    & ERE, NANOG, OCT, SOX, STAT \\ \hline
Oct4     & CP2, E2F, EBOX, ERE, NANOG, OCT, SOX, STAT \\ \hline
Sox2     & ERE, CP2, NANOG, OCT, SOX, STAT \\ \hline
Smad1    & ERE, CP2, NANOG, OCT, SOX, STAT \\ \hline
E2f1     & CP2, E2F, EBOX, OCT, STAT, ZF5 \\ \hline
Tcfcp2l1 & CP2, E2F, OCT, SOX, STAT \\ \hline
Zfx      & CP2, E2F, EBOX, OCT, ZF5 \\ \hline
Stat3    & E2F, EBOX, ERE, CP2, NANOG, OCT, SOX, STAT \\ \hline
Klf4     & E2F, EBOX, ERE, NANOG, OCT, SOX, STAT, ZF5 \\ \hline
Esrrb    & ERE, NANOG, OCT, SOX, STAT \\ \hline
c-Myc    & E2F, EBOX, ZF5 \\ \hline
n-Myc    & E2F, EBOX, OCT, STAT, ZF5 \\ \hline
p300     & ERE, CP2, NANOG, OCT, SOX, STAT \\ \hline
\end{tabular}
\end{threeparttable}
\end{table*}


TRANSFAC \cite{mybibfile:transfac} provides data on eukaryotic
transcription factors, their experimentally-proven binding sites,
consensus binding sequences (positional weight matrices) and
regulated genes. The nucleotide distribution matrix of aligned
binding sequences are provided in the TRANSFAC matrix. In the public
version database, $398$ matrices can be grouped into six categories
as vertebrates, insects, plants, fungi, nematodes and bacteria, and
$292$ of them are vertebrates used by our method.


The mouse genome GRCm38 \cite{mybibfile:musgen} is used to extract
sub-sequences corresponding to peak locations from ChIP-seq data.



\subsection{Area Under Curve}


In order to evaluate our method, we use the area under receiver
operating characteristic (ROC) curve (AUC)
\cite{mybibfile:meanofroc} to analysis our results.

In the ROC graph, $x$ axis represents false positive rate (FPR) and
$y$ axis represents true positive rate (TPR)
\cite{mybibfile:statistics}. Each point in the ROC space is
corresponding to a pair of (FPR, TPR). The curve is created by
plotting TPR against FPR at various threshold settings
\cite{mybibfile:introroc}. One score is yielded for a instance with
a probabilistic classifier, and higher score means a higher
probability. Therefore, a threshold can be set to transfer a
probabilistic classifier into a binary classifier with those scores
greater than threshold marked as yes instance and no instance in
else situation. Then, each threshold can get a point in the ROC
space, and the ROC curve of this classifier can be constructed by
linking all the points.

When a ROC curve for a classifier is plotted, there will be a closed
area between the curve and $x$ axis. Area under ROC curve (AUC) is
the portion of this area in the unit square. The value of AUC ranges
from $0$ to $1$, and a AUC value of $0.5$ equal to a random tagging.
Higher AUC value means a higher probability that the classifier is
scoring a positive instance greater than a negative instance, which
means that this classifier is more efficient and accurate.

Our method produces a motif score for each candidate vertebrate
motif in TRANSFAC database, and uses the ROC curve to evaluate the
performance of scoring motifs. We group a ranked list of vertebrate
TRANSFAC motifs corresponding to their factor families. All
vertebrate motifs in TRANSFAC database can be divided into the
positive set and the negative set, based on the current ChIP-seq
data. Then, we can plot the ROC curve using the ranked list of motif
families and calculate the AUC. Using AUC results, we can evaluate
our method and compare to other methods.


\subsection{Assessment of Ordered Adjacency Difference}

DNA motifs have different sequence-specific binding scores around
ChIPed peak points. On a specific ChIP-seq dataset, if a candidate
motif is a co-TF, it would enrich at the near area and disperse at
the remote area. We compare different distributions of
sequence-specific binding scores on two motifs of c-Myc, as shown in
Figure \ref{bscore_bar}. The motif V\$E2F\_03 (Figure
\ref{e2f_score_bar}) has good enrichment on $0$ to $30$ bins, and
sequence-specific binding scores of the near area are much higher
than the remote area. The motif V\$OCT1\_07 (Figure
\ref{oct_score_bar}) does not have significant changes between the
near area and the remote area.

\begin{figure*}[!htpb]
\centering \subfloat[V\$E2F\_03]{
\includegraphics[width=3.5in]{image/bscorebar_e2f}
\label{e2f_score_bar}} \subfloat[V\$OCT1\_07]{
\includegraphics[width=3.5in]{image/bscorebar_oct}
\label{oct_score_bar}} \hfil \caption{Comparison of
sequence-specific binding scores on two motifs.} \label{bscore_bar}
\end{figure*}


We use gamma distribution to weight the $f_1$ score, which can
enlarge scores at specific ranges near the origin and shrink scores
at the remote ranges. We compare different distributions of $f_1$
scores on two motifs of c-Myc, as shown in Figure \ref{dist_bar}.
V\$E2F\_03 (Figure \ref{e2f_bar}) is a co-TF on c-Myc, having clear
boundary between regions $0-15$ and regions $15-40$. Positive scores
locate in regions $0-15$ that gamma distribution values are large,
and negative scores are not too large to effect changes. V\$OCT1\_07
(Figure \ref{oct_bar}) are almost similar in all regions, and large
negative scores reduce the effect of changes between enrichment
regions and remote regions.

\begin{figure*}[!htpb]
\centering
\subfloat[V\$E2F\_03]{
\includegraphics[width=3.5in]{image/gammabar_e2f}
\label{e2f_bar}}
\subfloat[V\$OCT1\_07]{
\includegraphics[width=3.5in]{image/gammabar_oct}
\label{oct_bar}} \hfil \caption{Comparison of $f_1$ scores on two
motifs of c-Myc} \label{dist_bar}
\end{figure*}



In order to reflect distribution of the $f_1$ score, positive
changes enrich motif binding affinity, and negative changes lead to
opposite situation. We also compare different distributions of $f_2$
scores on two motifs of c-Myc, as shown in Figure \ref{f2_dist_bar}.
V\$E2F\_03 (Figure \ref{f2_e2f_bar}) has more positive scores than
negative scores, which enhance binding ability. V\$OCT1\_07 (Figure
\ref{f2_oct_bar}) has large negative scores in all regions.

\begin{figure*}[!htpb]
\centering \subfloat[V\$E2F\_03]{
\includegraphics[width=3.5in]{image/gammabar_e2f_f2}
\label{f2_e2f_bar}}
\subfloat[V\$OCT1\_07]{
\includegraphics[width=3.5in]{image/gammabar_oct_f2}
\label{f2_oct_bar}} \hfil \caption{Comparison of $f_2$ scores on two
motifs of c-Myc} \label{f2_dist_bar}
\end{figure*}



\subsection{Comparison to Existing Methods}

We evaluate the performance of our method on ChIP-seq data in ES
cells. Also, we compare to other three existing methods having good
performances on classifying co-TFs, CEAS \cite{mybibfile:ceas},
CORE\_TF \cite{mybibfile:coretf} and CENTDIST
\cite{mybibfile:centdist}. Our method obtains good results on
several datasets, but has some shortages on other datasets, as shown
in Table \ref{auc_result}.


CEAS (cis-regulatory element annotation system)
\cite{mybibfile:ceas} is a web server that can identify enriched
transcription factor-binding motifs from user-defined genome-scale
ChIP regions. CEAS uses several features, including sequence
retrieval, conservation plot, nearby gene mapping, motif finding and
enrichment analysis \cite{mybibfile:ceas}. CORE\_TF (Conserved and
Over-REpresented Transciption Factor binding
sites)\cite{mybibfile:coretf}, can identify common transcription
factor binding sites in promoters of co-regulated genes. CORE\_TF
finds experimental datasets for over represented PWMs from TRANSFAC
database, and a unique feature matchs the random set to the
experimental set of promoters by GC content. CENTDIST is a web based
co-motif scanning program \cite{mybibfile:centdist}. It does not
need user specific background and parameters being automatically
determined on the motif distribution around ChIP-seq peaks.


On $13$ ChIP-seq datasets, our method obtains best AUC results on
four datasets, E2F1 ($0.8828$), ESRRB ($0.8466$), KLF4 ($0.8818$)
and NMYC ($0.9432$). CENTDIST produces best AUC results on eight
datasets, CMYC ($0.9957$), NANOG ($0.9699$), OCT4 ($0.9300$), SMAD1
($0.9507$), SOX2 ($0.9507$), STAT3 ($0.9175$), TCFCP ($0.9072$) and
ZFX ($0.8758$). CORE\_TF promBG under promoter background uses
enriched regions with size $200$ gets the best AUC result on one
datasets, P300 ($0.9397$). AUC results of our method on $13$
datasets are all above $0.8$, and the most accurate result is
$0.9611$ archived on CMYC dataset.


\begin{table*}[!htpb]
\centering
\begin{threeparttable}
\renewcommand{\arraystretch}{1.3}
\caption{AUC results of our method, CENTDIST, CEAS, and CORE\_TF on $13$ ChIP-seq datasets}
\label{auc_result}
\begin{tabular}{*{12}{|l}|}
\hline \multirow{2}{*}{} & \multirow{2}{*}{our method} &
\multirow{2}{*}{CENTDIST} & \multicolumn{3}{c|}{CORE\_TF promBG
\tnote{a}} & \multicolumn{3}{c|}{CORE\_TF randBG \tnote{b}} &
\multicolumn{3}{c|}{CEAS \tnote{c}}\\
\cline{4-12}
& & & 200 & 400 & 1000 & 200 & 400 & 1000 & 200 & 400 & 1000\\
\hline
CMYC  & 0.9611 & \textbf{0.9957} & 0.9892 & 0.9742 & 0.9355 & 0.9742 & 0.9505 & 0.9097 & 0.7731 & 0.7828 & 0.5806\\
E2F1  & \textbf{0.8828} & 0.8761 & 0.8202 & 0.7966 & 0.7758 & 0.8076 & 0.7862 & 0.7303 & 0.5789 & 0.5625 & 0.5746\\
ESRRB & \textbf{0.8466} & 0.7869 & 0.6373 & 0.6627 & 0.6065 & 0.5359 & 0.5451 & 0.6183 & 0.6203 & 0.6072 & 0.6111\\
KLF4  & \textbf{0.8818} & 0.8550 & 0.7075 & 0.7058 & 0.6908 & 0.7058 & 0.6950 & 0.6813 & 0.6708 & 0.6883 & 0.6021\\
NANOG & 0.8807 & \textbf{0.9699} & 0.9320 & 0.9399 & 0.9020 & 0.9255 & 0.9046 & 0.8327 & 0.8386 & 0.8510 & 0.7268\\
NMYC  & \textbf{0.9432} & 0.8889 & 0.8052 & 0.7915 & 0.7627 & 0.7922 & 0.7719 & 0.7418 & 0.7255 & 0.6137 & 0.6039\\
OCT4  & 0.8286 & \textbf{0.9300} & 0.8767 & 0.8908 & 0.9067 & 0.8625 & 0.8342 & 0.7900 & 0.8650 & 0.8175 & 0.8017\\
P300  & 0.8667 & 0.8646 & \textbf{0.9397} & 0.9364 & 0.8657 & 0.8860 & 0.8169 & 0.7270 & 0.7917 & 0.7741 & 0.6184\\
SMAD1 & 0.8299 & \textbf{0.9507} & 0.9430 & 0.9287 & 0.8520 & 0.9364 & 0.9167 & 0.8191 & 0.7906 & 0.8531 & 0.7007\\
SOX2  & 0.8460 & \textbf{0.9507} & 0.9035 & 0.9068 & 0.8947 & 0.9145 & 0.8969 & 0.8235 & 0.8531 & 0.8448 & 0.8684\\
STAT3 & 0.8017 & \textbf{0.9175} & 0.8742 & 0.8525 & 0.7875 & 0.7892 & 0.7275 & 0.7300 & 0.8067 & 0.7513 & 0.7546\\
TCFCP & 0.8207 & \textbf{0.9072} & 0.6889 & 0.6719 & 0.5386 & 0.6627 & 0.6484 & 0.6641 & 0.6333 & 0.6144 & 0.6105\\
ZFX   & 0.8494 & \textbf{0.8758} & 0.8353 & 0.8248 & 0.7732 & 0.8288 & 0.8013 & 0.7190 & 0.6327 & 0.5137 & 0.5137\\
%\hline AVG & 0.8646 & \textbf{0.9053} & 0.8425 & 0.8371 & 0.7917 & 0.8170 & 0.7919 & 0.7528 & 0.7369 & 0.7134 & 0.6590\\
\hline
\end{tabular}
\begin{tablenotes}
\item [a] CORE\_TF promBG under promoter background uses enriched regions with size $200$, $400$ and $1000$;
\item [b] CORE\_TF randBG under random genome background uses enriched regions with size $200$, $400$ and $1000$;
\item [c] CEAS uses enriched regions with size $200$, $400$ and $1000$.
\end{tablenotes}
\end{threeparttable}
\end{table*}



Figure \ref{auc_figure} shows the comparison of best AUC results by
four methods on each dataset. We find that our method is more
accurate than other three methods on E2F1, ESRRB, KLF4 and NMYC
datasets. The average result of our method is better than those of
CORE\_TF and CEAS.


\begin{figure*}[!htpb]
\centering
\includegraphics[width=5in]{image/GeneFindingDraft}
\label{aucresult} \caption{Comparison of our method, CENTDIST, CEAS,
and CORE\_TF on ChIP-seq data in ES cells.} \label{auc_figure}
\end{figure*}




\section{Conclusion}


In order to predict co-associated TFs, we develop a novel method to
evaluate the associated situation between TFs. We design an
adjacency score for motif analysis, which can illustrate ordered
adjacency differences and co-TF binding affinity for candidate
motifs. On $13$ ChIP-seq datasets, our method obtains best AUC
results on four datasets, $0.8828$ for E2F1, $0.8466$ for ESRRB,
$0.8818$ for KLF4 and $0.9432$ for NMYC. AUC results of our method
on $13$ datasets are all above $0.8$, and the most accurate result
is $0.9611$ archived on CMYC.




\section*{Acknowledgments}
This work is supported by a grant from the National Science
Foundation of China (NSFC 61402326), Peiyang Scholar Program of
Tianjin University (no. 2016XRG-0009), and the Tianjin Research
Program of Application Foundation and Advanced Technology
(16JCQNJC00200).


\bibliographystyle{IEEEtran}
\bibliography{IEEEabrv,mybibfile}


%\begin{IEEEbiography}{Michael Shell}
%Biography text here.
%\end{IEEEbiography}
%
%\begin{IEEEbiographynophoto}{John Doe}
%Biography text here.
%\end{IEEEbiographynophoto}
%
%\begin{IEEEbiographynophoto}{Jane Doe}
%Biography text here.
%\end{IEEEbiographynophoto}

\end{document}
